exonum/src/proto/mod.rs

```diff
@@ -335,6 +337,77 @@ where
     }
 }
 
+impl<K, V> ProtobufConvert for exonum_merkledb::MapProof<K, V>
+where
+    K: BinaryKey,
+    V: BinaryValue,
```

Comment:
I'm not 100% sure whether using `BinaryValue` as a constraint similarly breaches the encapsulation; AFAIK, `BinaryValue` is automatically derived together with `ProtobufConvert`, so it *could* make sense to use `ProtobufConvert` constraint here as well.